src/Adapters/AuthorityBasedApprovalAuthorizer.php

```diff
@@ -4,20 +4,37 @@
 
 namespace ProfessionalWiki\PageApprovals\Adapters;
 
-use MediaWiki\Page\PageIdentity;
 use MediaWiki\Permissions\Authority;
 use ProfessionalWiki\PageApprovals\Application\ApprovalAuthorizer;
+use ProfessionalWiki\PageApprovals\Application\ApproverRepository;
+use Title;
+use WikiPage;
 
 class AuthorityBasedApprovalAuthorizer implements ApprovalAuthorizer {
 
 	public function __construct(
-		private Authority $authority
+		private Authority $authority,
+		private ApproverRepository $approverRepository
 	) {
 	}
 
-	public function canApprove( PageIdentity $page ): bool {
-		// TODO: logic
-		return true;
+	public function canApprove( WikiPage $page ): bool {
+		$sharedCategories = array_intersect(
+			$this->approverRepository->getApproverCategories( $this->authority->getUser()->getId() ),
+			$this->getPageCategories( $page )
+		);
+
+		return $sharedCategories !== [];
+	}
+
+	/**
+	 * @return string[]
+	 */
+	private function getPageCategories( WikiPage $page ): array {
+		return array_map(
+			fn( Title $categoryTitle ) => $categoryTitle->getText(),
+			iterator_to_array( $page->getCategories() )
+		);
```

Comment:
@waximabbax I'm moving the category case sensitivity discussion to a new thread.

What I described in https://github.com/ProfessionalWiki/PageApprovals/pull/33#discussion_r1659827129 is MediaWiki's behavior.

In our case we will need to normalize the categories we save (or retrieve). However, it won't be just as simple as lowercasing everything. Like @JeroenDeDauw mentioned we'll need to use MW's own normalization. Off the top of my head it does at least this:
* `Category:Foo` = `Category:foo` (First letter is case insensitive)
* `Category:Some thing` != `Category:Some Thing` (Rest of the letters are case sensitive)
* `Category:Some_thing` = `Category:Some thing` (Underscore is equal to space)
* Maybe some other special character handling

Reply:
My argument was that we ideally don't care about what MW normalization looks like and instead use MWs mechanisms for said normalization.

Reply:
We don't need to implement it manually, but we still need to normalize: https://github.com/ProfessionalWiki/PageApprovals/pull/51